store/sqlstore/upgrade.go

```diff
@@ -688,10 +688,10 @@ func UpgradeDatabaseToVersion511(sqlStore SqlStore) {
 
 func UpgradeDatabaseToVersion512(sqlStore SqlStore) {
 	if shouldPerformUpgrade(sqlStore, VERSION_5_11_0, VERSION_5_12_0) {
-		sqlStore.CreateColumnIfNotExistsNoDefault("TeamMembers", "SchemeGuest", "boolean", "boolean")
-		sqlStore.CreateColumnIfNotExistsNoDefault("ChannelMembers", "SchemeGuest", "boolean", "boolean")
-		sqlStore.CreateColumnIfNotExistsNoDefault("Schemes", "DefaultTeamGuestRole", "text", "VARCHAR(64)")
-		sqlStore.CreateColumnIfNotExistsNoDefault("Schemes", "DefaultChannelGuestRole", "text", "VARCHAR(64)")
+		sqlStore.CreateColumnIfNotExistsNoDefault("TeamMembers", "SchemeGuest", "tinyint(4)", "boolean")
```

Comment:
Retroactively updating this migration won't help anyone who's already upgraded to 5.12 or 5.13: should we move this to `UpgradeDatabaseToVersion516`? Or `UpgradeDatabaseToVersion514` and backport to the active release?

Reply:
@lieut-data - I've tried adding it to `upgrade.go` but it still resuls it:
```
Generating diff
210,211c210
<   PRIMARY KEY (`GroupId`,`ChannelId`),
<   KEY `idx_groupchannels_channelid` (`ChannelId`)
---
>   PRIMARY KEY (`GroupId`,`ChannelId`)
234,235c233
<   PRIMARY KEY (`GroupId`,`TeamId`),
<   KEY `idx_groupteams_teamid` (`TeamId`)
---
>   PRIMARY KEY (`GroupId`,`TeamId`)
```

Reply:
I admit I'm not sure how that could happen. cc @mkraft for clarity, but I maintain we need to do this in `UpgradeDatabaseToVersion514` for the reasons stated above.

Reply:
@lieut-data - you are correct of course. I've fixed up the order of the applied operations. It's fine now.

Reply:
@reflog, apologies for any confusion here -- but I'm still not seeing these indices being re-created in the `UpgradeDatabaseToVersion514`. The DB migration will pass now, but any customer that happened to do a fresh install of 5.10-5.13 won't have gotten these indexes, and won't ever get them unless we add them to a migration step.

Reply:
@lieut-data - the indices are added here in `upgrade.go`:
```go
func UpgradeDatabaseToVersion510(sqlStore SqlStore) {
	if shouldPerformUpgrade(sqlStore, VERSION_5_9_0, VERSION_5_10_0) {
		sqlStore.CreateColumnIfNotExistsNoDefault("Channels", "GroupConstrained", "tinyint(4)", "boolean")
		sqlStore.CreateColumnIfNotExistsNoDefault("Teams", "GroupConstrained", "tinyint(4)", "boolean")

		sqlStore.CreateIndexIfNotExists("idx_groupteams_teamid", "GroupTeams", "TeamId")
		sqlStore.CreateIndexIfNotExists("idx_groupchannels_channelid", "GroupChannels", "ChannelId")

		saveSchemaVersion(sqlStore, VERSION_5_10_0)
	}
}
```

Reply:
ok, spoke to @lieut-data , realized what I was missing. Added the indices to 514 migration to make sure they are performed.